external/zip/src/miniz.h

```diff
@@ -281,7 +281,7 @@
 #define MINIZ_LITTLE_ENDIAN 1
 #endif
 
-#if MINIZ_X86_OR_X64_CPU
+#if MINIZ_X86_OR_X64_CPU && !HERMES_UBSAN
```

Comment:
This is admittedly clumsy, but miniz does not allow overriding these defines from outside the miniz file, so I think a more principled change would require writing a lot more code.

Reply:
Alternatively, you could add a ubsan suppression annotation
```
__attribute__((no_sanitize("undefined")))
```
to any functions in this header that use unaligned accesses.
I think doing that would be fine to upstream to the miniz.h owner, and help other users of this library.

(replace `"undefined"` with a more specific annotation type like `"alignment"` if possible)

Reply:
I agree this would be a better fix, but we are all busy people :)

Reply:
Yeah, I don't need to be fixing miniz.

Reply:
Please do add this change to `external/zip/README.md` in case we update it, though.

Reply:
What's the purpose of the external/zip/sanitize-blacklist.txt? @avp @dulinriley

Reply:
Shouldn't we use `-fsanitize-blacklist=sanitize-blacklist`?

Reply:
Oh good point, I forgot I had already made that file.
Yes `-fsanitize-blacklist=sanitize-blacklist.txt` should suppress the problems from miniz.

Reply:
Thanks, for using `kuba--/zip`. Would be cool, if you can contribute back ;-)